sql/hivecontext-compatibility/src/test/scala/org/apache/spark/sql/hive/HiveContextCompatibilitySuite.scala

```diff
@@ -99,4 +105,41 @@ class HiveContextCompatibilitySuite extends SparkFunSuite with BeforeAndAfterEac
     assert(databases3.toSeq == Seq("default"))
   }
 
+  test("check change after refresh") {
```

Comment:
The test cases modified by this PR are used to verify `refreshTable` APIs. We also have test cases to verify the corresponding SQL interface, which is calling RefreshTable Command. For example, 
https://github.com/apache/spark/blob/d8a83a564ff3fd0281007adbf8aa3757da8a2c2b/sql/hive/src/test/scala/org/apache/spark/sql/hive/CachedTableSuite.scala#L164-L207

Now, to test the pure `HiveContext`, the only way we can do is to add a test case in `sql/hivecontext-compatibility`. 

Not sure if this can answer your question. Let me know if you have any concern